### ISSUE TITLE
A semicolon without a space gives unexpected error.

### ISSUE THREAD

**Issue by robertdebock** [ansible/ansible-lint#2226]
##### Summary

When a key-value pair misses a colon, ansible-lint complains about a further line.

##### Issue Type

- Bug Report

##### Ansible and Ansible Lint details

```console
ansible --version
ansible [core 2.12.2]
  config file = None
  configured module search path = ['/Users/my_user/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /opt/homebrew/lib/python3.9/site-packages/ansible
  ansible collection location = /Users/my_use/.ansible/collections:/usr/share/ansible/collections
  executable location = /opt/homebrew/bin/ansible
  python version = 3.9.13 (main, May 24 2022, 21:13:51) [Clang 13.1.6 (clang-1316.0.21.2)]
  jinja version = 3.0.3
  libyaml = True

ansible-lint --version
ansible-lint 6.3.0 using ansible 2.12.2
```

- ansible installation method: pip
- ansible-lint installation method: pip

##### OS / ENVIRONMENT

Nothing relevant to report.

##### STEPS TO REPRODUCE

With this playbook:

```yaml
---
- name: prove issue
  hosts: localhost
  become:no
  # Here `become:no` is missing a space, should be `become: no`.

  tasks:
    - name: sample task
      debug:
        msg: "sample ouput"
```

##### Desired Behavior

It would be nice if ansible-lint would report on the faulty line, `become:no` rather than a later line.

##### Actual Behavior

ansible-lint throws this warning:

```console
WARNING  Listing 1 violation(s) that are fatal
syntax-check: Ansible syntax check failed.
playbook.yml:1 ERROR! We were unable to read either as JSON nor YAML, these are the errors we got from each:
JSON: Expecting value: line 1 column 1 (char 0)

Syntax Error while loading YAML.
  could not find expected ':'

The error appears to be in '/private/tmp/issue/playbook.yml': line 6, column 3, but may
be elsewhere in the file depending on the exact syntax problem.

The offending line appears to be:


  tasks:
  ^ here
```

**Comment by ssbarnea** [comment]
That is invalid YAML, we cannot load the file. Not a bug.
